controller/cache/scan.go

```diff
@@ -625,9 +629,15 @@ func scannerDBChange(newVer string) {
 	}
 
 	// Skip if no auto-scan options are enabled
-	if !scanCfg.EnableAutoScanWorkload &&
-		!scanCfg.EnableAutoScanHost &&
-		!scanCfg.AutoScan {
+	enableAutoScanWorkload := false
+	if scanCfg.EnableAutoScanWorkload != nil {
+		enableAutoScanWorkload = *scanCfg.EnableAutoScanWorkload
+	}
+	enableAutoScanHost := false
+	if scanCfg.EnableAutoScanHost != nil {
+		enableAutoScanHost = *scanCfg.EnableAutoScanHost
+	}
+	if !enableAutoScanWorkload && !enableAutoScanHost && !scanCfg.AutoScan {
```

Comment:
This logic looks like incorrect. 
Let's say the operation sequence is:
1. scanCfg.AutoScan is set to true in 5.4.2
2. NV controller is upgraded to 5.4.3
3. Both enableAutoScanWorkload / enableAutoScanHost are explicitly set to false
It means "do not auto scan for workloads & hosts" but line 640 will be "if !false && !false && !true" and skip line 641 to trigger autoscan!
------
The initial value of local var enableAutoScanWorkloa / enableAutoScanHost should be from scanCfg.AutoScan.
(like in controller/cache/log.go)
When scanCfg.EnableAutoScanWorkload / scanCfg.EnableAutoScanHost is non-nil, their value is set accordingly. 
Then line 640~642 can simply be
```
if !enableAutoScanWorkload && !enableAutoScanHost {
    return
}
```